### DISCUSSION TITLE
QFIELD form traceability

### DISCUSSION THREAD

**Discussion by FGH33** [opengisch/QField#4091] [Q&A]
Hi everyone, I'm creating a Qfield project and I'm stuck trying to find out the way to set the traceability. Let's explain to see if someone can help me. I'm sure it's not hard but right now I can not find out how to do it. MAYBE THIS QUESTION IS ALREADY ASKED.

What I need is make a form which allows me collect information and update it later (monthly, yearly, etc.) avoiding delete the previous register. This allow me to get an historic register of when and why it was modified. I create one field type date name "last review" and another one name "next review". First one is filled automatically, the second one is filled by me. 

I sync the project and charge it in the app but when I make a change on an specific route (lines) it overwrite the previous register data instead of keep the first register and add the update. How can I fix it? 

Thank you !!!

**Comment by ThomasDV1984** [comment]
I think there is two ways: 
A normalized database with a 1:n relation
Or a history table. 
https://docs.qgis.org/3.28/en/docs/user_manual/working_with_vector/attribute_table.html#creating-one-or-many-to-many-relations

https://postgis.net/workshops/postgis-intro/history_tracking.html
